### ISSUE TITLE
SDK 1.2: sfb.devicesManager.selectedSpeaker.activeEndpoint null pointer bug

### ISSUE THREAD

**Issue by tigerdyret** [OfficeDev/skype-ios-app-sdk-samples#61]
In the SDK 1.1 I noticed a memory leak. This was resolved in 1.2, but now whenever I launch my app (both on ios 9 and ios 10+), I get a null pointer exception in ConversationHelper.m:
[self subscribeTo:_speaker selector:@selector(activeEndpoint)];

And hence sfb.devicesManager.selectedSpeaker.activeEndpoint = .loudspeaker
also fails.

Please fix asap since this is a very critical bug (the app is useless).

**Comment by Dove0412** [comment]
Same problem, very serious!

**Comment by Dove0412** [comment]
About ten percent of the success rate. Are there any new developments?

**Comment by sudhirnumino** [comment]
Getting the same issue as @tigerdyret with SDK v1.2, it works "sometimes".  Need a fix asap.

**Comment by christianlykke9** [comment]
+1
I am experiencing the same issue. I am currently stuck, so looking forward to a quick fix, thanks.

**Comment by SimonTob** [comment]
+1
I'm having the same issue. Please fix asap.

**Comment by jovillas** [comment]
Hi all, we have been tracking this bug and finally managed to reproduce it on a testing device. The problem is related to timing issues, that's the reason it "sometimes work" and the reason we were not able to find it.

We are working on a fix to prevent this situation or at least mitigate them. Will keep you posted on the progress.

Just out of curiosity... So far, we were only able to reproduce it on iPhone 6s (no simulator), are you seeing this same issue on other devices?

**Comment by sudhirnumino** [comment]
Had the same issue on iPad 4th Gen

**Comment by tigerdyret** [comment]
I have the same issue on an iPhone 5S (and 6S as you've already mentioned).

Thanks for the update!

**Comment by tigerdyret** [comment]
I am by the way able to reproduce the bug each time if I trigger/activate my app with a VoIP push notification followed by immediately initializing Skype and joining a meeting.

**Comment by halilyilmaz** [comment]
+1
I have same issue on an iPhone 6

**Comment by tigerdyret** [comment]
@jovillas do you have an ETA on this? I have a large danish client who has an app in production and this bug has rendered it rather useless.

Thank you for your support.

**Comment by halilyilmaz** [comment]
version updated. just released version 1.3 [https://www.microsoft.com/en-us/download/details.aspx?id=51962](link)

**Comment by jovillas** [comment]
Hi, a new version of the SDK was released this week and this issue should be fixed.

Thanks for the patience.

**Comment by VincentH-Net** [comment]
@tigerdyret where can I find the 1.0.3 release notes? What was fixed / changed? Thx. (https://msdn.microsoft.com/en-us/skype/appsdk/releasenotes does not mention 1.0.3)

**Comment by tigerdyret** [comment]
I'm not with Microsoft, so I don't know :-)

**Comment by jovillas** [comment]
There was no 1.0.3, we released 1.3 which you can find [here](https://msdn.microsoft.com/en-us/skype/appsdk/download).

The release notes you linked are for 1.3.

**Comment by VincentH-Net** [comment]
@jovillas Thx for your quick response; I mistyped - I meant 1.3. However my question remains: where can I find what was fixed / changed in 1.3? The [release notes page](https://msdn.microsoft.com/en-us/skype/appsdk/releasenotes) says: **Last Updated: 2/16/2017**; nothing specific to 1.3 is in there - It had the same content for the previous point release.

I expected the release notes page to be updated with a point release (and to mention the version number). Could they be updated?

A list of bug fixes would be helpful to determine whether/when to upgrade the apps that I am responsible for.
